owl/SOMA-IO.owl

```diff
@@ -73,6 +453,97 @@
+        <rdfs:label xml:lang="en">File Format</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#FormalLanguage -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#FormalLanguage">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Language"/>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#givesMeaningTo"/>
+                <owl:allValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Structured_Text"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:comment xml:lang="en">A Formal Language consists of words whose letters are taken from an alphabet and are well-formed according to a specific set of rules.
+
+From Wikipedia: https://en.wikipedia.org/wiki/Formal_language</rdfs:comment>
+        <rdfs:label xml:lang="en">Formal Language</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#GraphDatabase -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#GraphDatabase">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Database"/>
+        <rdfs:comment xml:lang="en">A Graph Database is a Database that uses graph structures for semantic queries with nodes, edges, and properties to represent and store data.</rdfs:comment>
+        <rdfs:label xml:lang="en">Graph database</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#HostRole -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#HostRole">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#InterfaceComponentRole"/>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ontologydesignpatterns.org/ont/dul/DUL.owl#isDefinedIn"/>
+                <owl:someValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#PluginSpecification"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:label xml:lang="en">Host role</rdfs:label>
+    </owl:Class>
+    
+
+
+    <!-- http://www.ease-crc.org/ont/SOMA.owl#Human-readable_Programming_Language -->
+
+    <owl:Class rdf:about="http://www.ease-crc.org/ont/SOMA.owl#Human-readable_Programming_Language">
+        <rdfs:subClassOf rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Programming_Language"/>
+        <rdfs:subClassOf>
+            <owl:Restriction>
+                <owl:onProperty rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#givesMeaningTo"/>
+                <owl:allValuesFrom rdf:resource="http://www.ease-crc.org/ont/SOMA.owl#Source_Code"/>
+            </owl:Restriction>
+        </rdfs:subClassOf>
+        <rdfs:comment xml:lang="en">A Programming language like Java, Python etc. but not binary machine code.</rdfs:comment>
```

Comment:
Where would assembly fall?

Reply:
Currently, just under the umbrella of Programming Language. One could introduce the concept `Low-level programming language`.